### PR TITLE
Add helper function ``area`` and ``layout_from_area_density`` & allow auto chunks

### PR DESCRIPTION
**Pull request by FabianHofmann**
- add functions `area` & `layout_from_area_density`
 - allow for `auto` chunks when loading cutout

## Motivation and Context

The ``area`` function is helpful for custom workflows, the same with the new layout function.

The auto chunking **can** help to accelerate the computation of the conversion functions. 

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [x] I tested my contribution locally and it seems to work fine.
- [x] I locally ran `pytest` inside the repository and no unexpected problems came up.
- [x] I have adjusted the docstrings in the code appropriately.
- [x] I have documented the effects of my code changes in the documentation `doc/`.
- [x] I have added newly introduced dependencies to `environment.yaml` file.
- [x] I have added a note to release notes `doc/release_notes.rst`.
- [x] I have used `pre-commit run --all` to lint/format/check my contribution

### CODE REVIEW

File: atlite/cutout.py [hunk 1]
```diff
@@ -564,6 +567,27 @@ def intersectionmatrix(self, shapes, shapes_crs=4326):
         """
         return compute_intersectionmatrix(self.grid, shapes, self.crs, shapes_crs)
 
+    def area(self, crs=3035):
```
Comment (euronion) [inline atlite/cutout.py:570]: I don't like the default `crs=3035` - it's only accurate for Europe.

By default I think it should use the CRS of the cutout or throw an error (asking the user to specify a CRS).

File: atlite/cutout.py [hunk 2]
```diff
@@ -618,6 +642,27 @@ def layout_from_capacity_list(self, data, col="Capacity"):
...
+        """
+        Get a capacity layout from a uniform capacity density.
+
+        Parameters
+        ----------
+        capacity_density : float
+            Capacity density in capacity/km^2
+        crs : int, optional
+            CRS to calculate the total area of grid cells in m^2.
+            The default is 3035 which is suitable for the European area.
```
Comment (euronion) [inline atlite/cutout.py:655]: See comment in the `area` function on (default) CRS above .

File: atlite/cutout.py [hunk 3]
```diff
@@ -618,6 +642,27 @@ def layout_from_capacity_list(self, data, col="Capacity"):
         )
         return data.to_xarray().reindex_like(self.data).fillna(0)
 
+    def layout_from_area_density(self, capacity_density, crs=3035):
+        """
+        Get a capacity layout from a uniform capacity density.
```
Comment (euronion) [inline atlite/cutout.py:647]: I find the formulation "from density" a bit confusing. What do you think about `uniform_density_layout(...)` instead, to be consistent with the `uniform_layout(...)`?

File: atlite/cutout.py [hunk 4]
```diff
@@ -150,7 +150,10 @@ def __init__(self, path, **cutoutparams):
 
         path = Path(path).with_suffix(".nc")
         chunks = cutoutparams.pop("chunks", {"time": 100})
-        storable_chunks = {f"chunksize_{k}": v for k, v in (chunks or {}).items()}
+        if isinstance(chunks, dict):
+            storable_chunks = {f"chunksize_{k}": v for k, v in (chunks or {}).items()}
```
Comment (euronion) [inline atlite/cutout.py:154]: Is auto-chunking stable enough nowadays to become the default?

**Review by euronion — CHANGES_REQUESTED** [review]
Minor comments

File: atlite/cutout.py (hunk 4 above)
Comment (FabianHofmann) [reply atlite/cutout.py:154]: I don't think we should set it as default yet. There might still be some problems with `atlite` functionalities like reprojecting, which do not fully work with chunked spatial dimensions.

File: atlite/cutout.py (hunk 1 above)
Comment (FabianHofmann) [reply atlite/cutout.py]: This actually a good point, thanks for raising. I adjusted the code accordingly.

File: atlite/cutout.py (hunk 4 above)
Comment (coroa) [reply atlite/cutout.py:154]: `(chunks or {})` is equivalent to `chunks` (since `chunks` is a dict)

**Pull request merged**